aten/src/ATen/native/DispatchStub.h

```diff
@@ -183,6 +184,7 @@ struct TORCH_API DispatchStubImpl {
     void* cuda_dispatch_ptr = nullptr;
     void* hip_dispatch_ptr = nullptr;
     void* mps_dispatch_ptr = nullptr;
+    void* xpu_dispatch_ptr = nullptr;
```

Comment:
For struct size could we also guard this with USE_XPU ?
We can also guard set_xpu_dispatch_ptr with this and make it either an error if called outside of an XPU build